### ISSUE TITLE
Ability to swipe pictures in fullscreen 

### ISSUE THREAD

**Issue by Pafzedog** [jhass/insporation#45]
Hello,

It will be very usefull to swipe pictures in fullscreen mode.

* go to a post with several images
* touch one -> picture is displayed fullscreen
* swipe left or right : nothing happens

**Comment by jhass** [comment]
Fullscreen allows zooming and thus lateral movement into any direction. I'm not a big fan of having dual use gestures that cause different actions based on some state ( so on this example being zoomed in and close to the edge of the image or being zoomed in or not). My experience is this is often confusing and causing unintended action (here again being zoomed in I might want to move to the edge of the image quickly but not to the next one and loose my zoom level).

**Comment by Pafzedog** [comment]
I totally agree with you that zooming in is a very convenient fonctionnality and that inintentionnal actions have to be avoid but when I read a post with about thirty pictures it is very annoying to repeat those steps again and again : touch for fullscreen, touch to exit, swipe to next, touch to fullscreen, etc ... (I have a 5.1' screen and behind my forties lol)
so... won't it be possible to allow "swipe to next" only when fullscreen is not in zoom state ?

**Comment by jhass** [comment]
It's probably possible but in my experience fairly tricky to implement, especially in a way that feels like good UX. 

Personally I tend to swipe on the post and only tap to fullscreen if something is really interesting me, I very rarely want to see each image in fullscreen. 

Maybe we can take a page from desktop and implement a slider on the bottom that receives horizontal swipe gestures to flip through the images. It could then also fade out when the image is zoomed in and only return on zoom out.

**Comment by Pafzedog** [comment]
Honestly, I don't know either my or your experience is the most representative, it would be hard to guess without other users' feedback.
My feeling is that I like to use an app when I can interact in an intuitive way and in this case, this is not what I feel.
Sorry, I don't want to be rude but I can't really explain myself in english :s

**Comment by jhass** [comment]
Let's hear some more opinions :)

It could also be interesting to research how some other (popular) apps tackle this problem space. There's not always the best solution to find as you often have people in these companies that change something to change something, but I often found it useful for inspiration and challenging my own opinion.

**Comment by Pafzedog** [comment]
Fedilab (https://framagit.org/tom79/fedilab) or Signal seem to handle this the same way: 
* touch to fullscreen
* swipe horizontally to next picture
* you can pinch to zoom in and out
* swipe horizontally when zoomed  **and** close to the edge switch to next picture (but when you go back to previous, zoom level is preserved)

**Comment by tclaus** [comment]
I quickly created a post with multiple images and immediately stumbled on this behaviour:
On zoomed fullscreen it would be nice if I can move the image to a border and change to the next. 

Not a must, but worth to investigate imho.

**Comment by paskalito** [comment]
+1 For the ability to swipe through pics in full screen. (its my main usecase)

About the UI Problem > what @Pafzedog wrote is the ui standard Imho 

1. One clarification: If you zoom in (more than 1-10% (felt) and then you swipe left - no matter how long or fast - it will only get you to where the picture ends and stop there. 
Only if are at the very edge of the picture and then swipe a second time against "the (virtual) wall" you will go to the next picture.

2.  i find doubletap gestures really handy as in: 
1st time = 100% zoom, 2nd time = 200% zoom, 3rd time = 0% Zoom
(also as a way to get our if you pinched the picture)

3.  I'd love to be able to hold on a picture and get a context menu: Save and share (The actual picture). Because right now i need to make a screenshot in order to send someone a picture i find in d*

**Comment by jhass** [comment]
Let's track and discuss picture context actions on a separate issue please to stay focused :)
